tests/momento/simple_cache_client/test_scalar_async.py

```diff
@@ -9,274 +12,268 @@
-    key = uuid_bytes()
-    value = uuid_bytes()
-
-    set_resp = await client_async.set(cache_name, key, value)
-    assert isinstance(set_resp, CacheSet.Success)
-
-    get_resp = await client_async.get(cache_name, key)
-    assert isinstance(get_resp, CacheGet.Hit)
-    assert get_resp.value_bytes == value
-
-
-async def test_get_returns_miss(client_async: SimpleCacheClientAsync, cache_name: str) -> None:
-    key = uuid_str()
-
-    get_resp = await client_async.get(cache_name, key)
-    assert isinstance(get_resp, CacheGet.Miss)
-
-
-async def test_expires_items_after_ttl(client_async: SimpleCacheClientAsync, cache_name: str) -> None:
-    key = uuid_str()
-    val = uuid_str()
-
-    await client_async.set(cache_name, key, val, timedelta(seconds=2))
-    get_response = await client_async.get(cache_name, key)
-    assert isinstance(get_response, CacheGet.Hit)
-
-    time.sleep(4)
-    get_response = await client_async.get(cache_name, key)
-    assert isinstance(get_response, CacheGet.Miss)
-
-
-async def test_set_with_different_ttl(client_async: SimpleCacheClientAsync, cache_name: str) -> None:
-    key1 = uuid_str()
-    key2 = uuid_str()
-
-    await client_async.set(cache_name, key1, "1", timedelta(seconds=2))
-    await client_async.set(cache_name, key2, "2")
-
-    # Before
-    get_response = await client_async.get(cache_name, key1)
-    assert isinstance(get_response, CacheGet.Hit)
-    get_response = await client_async.get(cache_name, key2)
-    assert isinstance(get_response, CacheGet.Hit)
+def a_cache_name_validator():
+    async def with_non_existent_cache_name_it_throws_not_found(
+        client_async: SimpleCacheClientAsync, cache_name_validator
+    ) -> None:
+        cache_name = uuid_str()
+        response = await cache_name_validator(cache_name)
+        assert response.error_code == MomentoErrorCode.NOT_FOUND_ERROR
+
+    async def with_null_cache_name_it_throws_exception(
+        client_async: SimpleCacheClientAsync, cache_name_validator
+    ) -> None:
+        response = await cache_name_validator(None)
+        assert response.error_code == MomentoErrorCode.INVALID_ARGUMENT_ERROR
+        assert response.inner_exception.message == "Cache name must be a non-empty string"
+
+    async def with_empty_cache_name_it_throws_exception(
+        client_async: SimpleCacheClientAsync, cache_name_validator
+    ) -> None:
+        response = await cache_name_validator("")
+        assert response.error_code == MomentoErrorCode.INVALID_ARGUMENT_ERROR
+        assert response.inner_exception.message == "Cache header is empty"
+
+    async def with_bad_cache_name_throws_exception(client_async: SimpleCacheClientAsync, cache_name_validator) -> None:
+        response = await cache_name_validator(1)
+        assert response.error_code == MomentoErrorCode.INVALID_ARGUMENT_ERROR
+        assert response.inner_exception.message == "Cache name must be a non-empty string"
```

Comment:
I was going to enable mypy checks on the tests shortly. The functions in this file (and the other) will need return types for all the functions. Most of them return None, though others will return the response types.

Reply:
This PR is just moving existing functions around, so we'll have type problems one way or another. Can we handle that in a different PR?

Reply:
I'll see what I can puzzle out.

Reply:
Using `key_validator` as an example, `key_validator` is returning a `<function describe_get.<locals>.key_validator.<locals>._key_validator at 0x10df21950>` and the internal `_key_validator` is returning a `<coroutine object SimpleCacheClientAsync.get at 0x10df25200>`.

I have no idea what that signature might be.

Reply:
This is over my head. I'd like to get this in and then work on the types.